### PR TITLE
LPD-28764 Migrate KBArticleSchedule testcase to integration test

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
LPD-28764 Migrate Update Test to integration test

# What is this trying to solve?

https://liferay.atlassian.net/browse/LPD-28764

Some POSHI test of KBArticle Schedule are [flaky](https://liferay.atlassian.net/browse/LPD-28228) and others can be migrated to integration test

# How am I fixing it?

I'm migrating some test when we update the KBArticle. On this [case](https://liferay.atlassian.net/browse/LPD-28228) we changed the service to allow scheduled articles to be imported/exported.

# How can you verify that it works?

Run the included automated tests.

# Commits

- **LPD-28764 migrate POSHI test to integration test to tests that ensures that users can edit a scheduled article.**
- **LPD-28764 migrate POSHI test to integration test that ensures that users can publish a draft article with a scheduled date.**
- **LPD-28764 modify service to allow export by several statuses**
- **LPD-28764 add method to get the latest KB article with several statutes**
- **LPD-28764 buildService**
- **LPD-28764 baseline**
- **LPD-28764 obtain the parent article from all the exportable statuses**
- **LPD-28764 migrate POSHI test to ensure that the scheduled article can be published in the live site after reaching the scheduled date.**
- **LPD-28764 do not do certain actions if we are in delete site process, this prevent the 'Unable to delete data for portlet null in group' on tests**
- **LPD-28764 extract common code by default**
- **LPD-28764 migrate POSHI test to integration test, this test ensures that the expired article can be published again when the date reaches a new scheduled date.**

### CODE REVIEW

**Comment by AliciaGarciaGarcia** [comment]
ci:test:knowledge-base

**Comment by AliciaGarciaGarcia** [comment]
ci:test:content-management

**Comment by AliciaGarciaGarcia** [comment]
ci:report:20998386

**Comment by AliciaGarciaGarcia** [comment]
ci:report:21099395

**Comment by jkappler** [comment]
@AliciaGarciaGarcia, only the small comment, otherwise, all looks great ❤️.

File: modules/apps/knowledge-base/knowledge-base-test/src/testIntegration/java/com/liferay/knowledge/base/service/test/KBArticleLocalServiceTest.java [hunk 1]
```diff
@@ -105,14 +105,14 @@ public void setUp() throws Exception {
 
 		_serviceContext = ServiceContextTestUtil.getServiceContext(
 			_group, _user.getUserId());
+
+		_serviceContext.setWorkflowAction(WorkflowConstants.ACTION_PUBLISH);
```
[the remark this answers is not in the document]
Comment (AliciaGarciaGarcia) [reply modules/apps/knowledge-base/knowledge-base-test/src/testIntegration/java/com/liferay/knowledge/base/service/test/KBArticleLocalServiceTest.java:109]: ok, I will drop the commit that I thought that was an improve.

Thanks!!

I will drop and send

**Comment by AliciaGarciaGarcia** [comment]
rebased

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:content-management

**Comment by AliciaGarciaGarcia** [comment]
ups, wrong button

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Review by jkappler — APPROVED** [review]

**Pull request closed without merging**